doc/developer-guides/c_coding_guidelines.rst

```diff
@@ -3441,6 +3441,74 @@ Compliant example::
        };
 
 
+C-NC-08: Function name shall be descriptive
+===========================================
+
+Function name shall be descriptive and clearly indicate the purpose of the
+function. Some detailed rules are listed below:
+
+1) If the function is doing actions, it shall be named with one of the following
+   formats:
+
+   a) <verb>_<nouns>, such as 'init_vmcs'.
+   b) <verb>_<adjective>_<nouns>, such as 'init_primary_pcpu'.
+2) If the function is doing checks, it shall be named with one of the following
+   formats:
+
+   a) is_<nouns>, such as 'is_space'.
+   b) is_<nouns>_<adjective>, such as 'is_pcpu_active'.
+3) If the function is doing conversions, it shall be named with one of the
+   following formats:
+
+   a) <nouns>_to_<nouns>, such as 'irq_to_vector'.
+   b) <nouns>2<nouns>, such as 'gva2gpa'.
+4) If the function is specific for one module and the name is not descriptive
+   enough with prior rules, it shall be named with the module name as prefix, such
+   as 'vie_read_mmio'.
+5) If the function is a wrapper of inline Assembly codes, it shall be named with
+   'asm' as prefix, such as 'asm_pause'.
```

Comment:
Is ``pause`` a valid function name (it's just a verb without a noun).  Should this be ``asm_pause_something`` following the rules above for ``<verb>_<nouns>`` ?

Reply:
> Is `pause` a valid function name (it's just a verb without a noun). Should this be `asm_pause_something` following the rules above for `<verb>_<nouns>` ?

@dbkinder 
`pause` is an Assembly instruction mnemonic. It is machine code which targets processor directly.
I update this rule in new version. Could you please help review it again?
Thanks!